src/prefect/orion/schemas/states.py

```diff
@@ -64,108 +62,6 @@ class Config:
     )
     state_details: StateDetails = Field(default_factory=StateDetails)
 
-    @overload
-    def result(state_or_future: "State[R]", raise_on_failure: bool = True) -> R:
-        ...
-
-    @overload
-    def result(
-        state_or_future: "State[R]", raise_on_failure: bool = False
-    ) -> Union[R, Exception]:
-        ...
-
-    def result(self, raise_on_failure: bool = True):
```

Comment:
Should I make this backwards compatible by importing from the client-side dynamically? I this I ought to with a warning.

Reply:
Agreed, seems like we should have it backwards compatible for at least a single release.

Reply:
Done